### PR TITLE
Plotting review

### PR DESCRIPTION
**Pull request by hemmelig**
## What is the purpose of this Pull Request? 
Review the event and trigger summary plotting routines. Most of these changes were purely stylistic, but a few places I have introduced some additional information

- Add custom DateFormatter for plotting times to arbitrary sub-second precision (d01e1ca)
- Review of LUT plotting  (29012bf)
- Review of the event summary plotting (ac61068)
- Review of the trigger summary plotting (b047c30)

More information on the specific changes can be found in each commit message.

In addition, move the setting of the default colormap to viridis to the top level of the QMigrate.plot module.

Note: This will require some careful manual merging with development after #78 -> handled when handling merge conflicts
Note: Will also require some updates to the keys used when accessing the triggered events dataframes - `COA_V` -> `TRIG_COA` in the `_plot_event_scatter` function.

## Test cases
Ran all examples - no issues.

- [x] All examples run without any new warnings
- [x] test_benchmarks.py reports all example tests pass

### System details
Please state the systems on which you have tested this change.
Ubuntu 18.04

### Final checklist
- [x] `develop` base branch selected?
- [x] All tests still pass.
- [x] Any new or changed features are fully documented.

### CODE REVIEW

File: QMigrate/plot/event.py [hunk 1]
```diff
@@ -140,7 +140,8 @@ def _plot_waveform_gather(ax, lut, event, idx):
     mint_i, maxt_i = [np.argmin(abs(times_utc - t)) for t in (mint, maxt)]
     times_plot = event.data.times(type="matplotlib")[mint_i:maxt_i]
     for i, signal in enumerate(np.rollaxis(event.data.filtered_signal, 1)):
-        for data, c, comp in zip(signal, WAVEFORM_COLOURS1, "ENZ"):
+        for data, c, comp in zip(signal[::-1], WAVEFORM_COLOURS1[::-1],
+                                 "ENZ"[::-1]):
```
Comment (hemmelig) [inline QMigrate/plot/event.py:144]:
```suggestion
        for data, c, comp in zip(signal[::-1], WAVEFORM_COLOURS1, "ZNE"):
```
Could do the above and re-order the colours in `WAVEFORM_COLOURS1` - not critical.

File: QMigrate/lut/lut.py [hunk 2]
```diff
@@ -557,70 +615,127 @@ def plot(self, fig, gs, slices=None, hypocentre=None, station_clr="k"):
...
+        grid_min, grid_max = self.get_grid_extent(cells=True)
+
+        min_extent = [min(a, b) for a, b in zip(stat_min, grid_min)]
+        max_extent = [max(a, b) for a, b in zip(stat_max, grid_max)]
+        diff = abs(np.subtract(max_extent, min_extent))
+
+        min_extent = np.subtract(min_extent, 0.05*diff)
+        max_extent = np.add(max_extent, 0.05 * diff)
+
+        return self.coord2grid([min_extent, max_extent], inverse=True)
```
Comment (twinder36) [inline QMigrate/lut/lut.py:716]:
```suggestion
        return np.array([min_extent, max_extent])
```

File: QMigrate/plot/trigger.py [hunk 3]
```diff
@@ -66,120 +66,88 @@ def trigger_summary(events, starttime, endtime, run, marginal_window,
...
-        axes[4].scatter(x, z, s=50, c=c, cmap="viridis", vmin=vmin, vmax=vmax)
-        axes[5].scatter(z, y, s=50, c=c, cmap="viridis", vmin=vmin, vmax=vmax)
-
-        # --- Add colourbar ---
-        spec = GridSpec(9, 18).new_subplotspec((2, 7), colspan=1, rowspan=5)
-        cax = fig.add_subplot(spec)
-        cax.set_axis_off()
-        cbar = fig.colorbar(sc, ax=cax, orientation="vertical", fraction=0.4)
-        cbar.ax.set_ylabel("Peak coalescence value", rotation=90, fontsize=14)
+                        label="Trigger threshold")
```
Comment (twinder36) [inline QMigrate/plot/trigger.py:118]: Hmm maybe this should stay as "Detection threshold" given that's the parameter name / sort of what it is. I made this change late last night..

Comment (hemmelig) [inline QMigrate/plot/trigger.py:118]:
```suggestion
                         label="Detection threshold") 
```

**Pull request merged**